File: src/common-regex.ts
import { regex } from "./primitives";

export const whitespace = regex(/^\s+/, "whitespace");
export const digits = regex(/^\d+/, "digits");
export const letters = regex(/^[a-zA-Z]+/, "letters");
export const word = regex(/^\w+/, "word");
export const integer = regex(/^-?\d+/, "integer");
export const float = regex(/^-?\d*\.\d+/, "float");
export const identifier = regex(/^[a-zA-Z_][a-zA-Z0-9_]*/, "identifier");

File: parso/common-regex.py
import re
from .primitives import regex

whitespace = regex(re.compile(r"\s+"), "whitespace")
number = regex(re.compile(r"-?\d+(\.\d+)?([eE][+-]?\d+)?"), "number")
integer = regex(re.compile(r"-?\d+"), "integer")
identifier = regex(re.compile(r"[a-zA-Z_][a-zA-Z0-9_]*"), "identifier")
quoted_string = regex(re.compile(r'"(?:\\.|[^"\\])*"'), "quoted string")
bool_true = regex(re.compile(r"true"), "true")
bool_false = regex(re.compile(r"false"), "false")
null = regex(re.compile(r"null"), "null")
